src/main/java/Aly/Aly.java

```diff
@@ -0,0 +1,451 @@
+        try {
+            File file = new File(fileName);
+            Scanner scanner = new Scanner(file);
+            int index = 0;
+
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+
+                String[] taskDetails = line.split("\\|");
+
+                switch (taskDetails[0]) {
+                case "T":
+                    taskList.add(new Todo(taskDetails[2].trim()));
+                    taskList.get(index).setDone(taskDetails[1].trim().equals("1"));
+                    break;
+                case "D":
+                    taskList.add(new Deadline(taskDetails[2].trim(), taskDetails[3].trim()));
+                    taskList.get(index).setDone(taskDetails[1].trim().equals("1"));
+                    break;
+                case "E":
+                    taskList.add(new Event(taskDetails[2].trim(), taskDetails[3].trim(), taskDetails[4].trim()));
+                    taskList.get(index).setDone(taskDetails[1].trim().equals("1"));
+                    break;
+                default:
+                    throw new IllegalArgumentException("Idk what task is this siah: " + taskDetails[0]);
+                }
+
+                index++;
+            }
+
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("Your file dont have lah... maybe because: " + e.getMessage());
+        } catch (Exception e) {
+            System.out.println("Error! Your file sucks because: " + e.getMessage());
+        }
+    }
+
+    private static void writeTasksToFile(String fileName) {
+        try {
+            FileWriter fileWriter = new FileWriter(fileName);
+
+            for (Task task : taskList) {
+                if (task != null) {
+                    String toSave = task.toString();
+                    char status = '0';
+                    if (toSave.charAt(4) == 'X') {
+                        status = '1';
+                    }
+                    char type = toSave.charAt(1);
+                    switch (type) {
+                    case 'T':
+                        int endIndex = 6;
+                        toSave = toSave.charAt(1) + "|" + status + "|" + toSave.substring(endIndex);
+                        break;
+                    case 'D':
+                        int byIndex = toSave.indexOf("(by: ");
+                        toSave = toSave.charAt(1) + "|" + status + "|" + toSave.substring(6, byIndex-1) + "|" + toSave.substring(byIndex+5,toSave.length()-1);
+                        break;
+                    case 'E':
+                        int fromIndex = toSave.indexOf("(from: ");
+                        int toIndex = toSave.indexOf("to: ");
+                        toSave = toSave.charAt(1) + "|" + status + "|" + toSave.substring(6,fromIndex-1) + "|" + toSave.substring(fromIndex+7,toIndex) + "|" + toSave.substring(toIndex+4,toSave.length()-1);
+                        break;
```

Comment:
Consider adding a new method to the class Task, inherit and override them Todo, Event, Deadline to ensure all the string formatting related to a class is within that class itself.